kernel/kernel-api/src/main/java/io/delta/kernel/DataWriteContext.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import java.util.List;
+import java.util.Map;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+
+/**
+ * Contains the context for writing data related to a partition to Delta table.
```

Comment:
Nit - generalize for partitioned/unpartitioned tables?

Reply:
I did one more update to class. The connector doesn't actually need to know the partition values. All we need is the connector to pass this object back to the Kernel. This avoids the complexity of case sensitive issues you raised on the insert PR.

Reply:
Just double checking my understanding, the connector only would need the partition values for the target directory path which we abstract within `getTargetDirectory()`?

Reply:
We return `DataWriteContextImpl`, which extends `DataWriteContext.` The partition values will be within the `DataWriteContextImpl`.